### DISCUSSION TITLE
Ability to get 1-off stacktrace in Ray output

### DISCUSSION THREAD

**Discussion by robertgdev** [spatie/ray#677] [Ideas]
Let me prefix this by saying that I would understand if your reply will be "not an issue, won't fix" ... 

Let me explain: I'm in the process of refactoring/upgrading a large legacy application that is making heavy use of Ajax calls ... ray is proving to be a life-saver here, so thank you for that. 

My issue is this: our architect has demanded that we wrap the ray() call in a simple debug wrapper (not unreasonable since this wrapper contains a number of other useful debug-related methods) which now means that instead of "ray()" we call "Debug::ray()" ... this works just fine, but the result is that the location of the ray call is now always displayed as "Debug.php:123" rather than the actual line where the call is to "Debug::ray()" is placed. 

Is there any way to tell ray to show not the top item from the stack trace but the top-minus-1 line from the stack trace so that the ray output would show where the actual call was placed? 

Thank you

**Comment by robertgdev** [comment]
Ping ... any ideas on this? Thank you!

**Reply by freekmurze** [reply]
Hi!

Because you're the only one requesting this, we're currently not investing time in this feature.

Should it be easy to add easily, feel free to send us a PR with tests and updated docs.

**Comment by robertgdev** [comment]
OK, thank you, I will try to take a look once I have some time available
(it may be 2-3 weeks before I get around to this).

**Comment by jmslbam** [comment]
Hi @rgasch,

I have the same thing in an application of mine.

Did you solve it yet? 

Do you have the filepath and lineNumber inside that general debug wrapper?
Because then you could maybe write a custom Payload like this one here? https://github.com/jmslbam/ray-php-errors and then use `ray()->sendRequest( $payload )` to send the data?

**Reply by robertgdev** [reply]
No, I decided to get rid of our wrapper, it was the quick-and-dirty solution ...
